src/model/priors/single_cell_bulk.rs

```diff
@@ -142,14 +142,12 @@ impl PairModel<DiscreteAlleleFreqs, ContinuousAlleleFreqs> for SingleCellBulkMod
         };
         let k_single = 0..n_obs_s + 1;
 
-        let k_start = if *af_bulk.start == 0.0 { // 0 as a bulk range start is always inclusive
-            0 as u64
-        } else {
-            // any non-zero bulk range start is always exclusive
-            (*af_bulk.start * n_obs_bulk as f64).floor() as u64 + 1
-        };
-        // any bulk range end is always inclusive
-        let k_end = (*af_bulk.end * n_obs_bulk as f64).floor() as u64 + 1;
+        // stable rust ranges currently only allow inclusive start points, for exclusive add
+        // 0.00000001 in the Event definition
+        let k_start = (*af_bulk.start * n_obs_bulk as f64).ceil() as u64;
+        // stable rust ranges currently only allow exclusive end points, for including an end point,
+        // add 0.00000001 in the Event definition
+        let k_end = (*af_bulk.end * n_obs_bulk as f64).floor() as u64;
```

Comment:
The trouble is, that ProSolo defines both inclusive AND exclusive start points, and both inclusive AND exclusive end points for ranges (starting here: https://github.com/PROSIC/prosolo/blob/master/src/call.rs#L81). Thus until we have ranges that can pass this information on to this point, I don't see how I could handle this here. But I do agree that the current solution isn't pretty and I'd prefer treating this here!

And in any case, thanks for the thorough code review!